lib/features/family/features/parent_giving_flow/presentation/pages/parent_amount_page.dart

```diff
@@ -74,44 +84,97 @@ class _ParentAmountPageState extends State<ParentAmountPage> {
-      ),
-      floatingActionButtonLocation: FloatingActionButtonLocation.centerFloat,
-      floatingActionButton: Padding(
-        padding: const EdgeInsets.symmetric(horizontal: 24),
-        child: FunButton(
-          onTap: widget.authcheck
-              ? () async {
-                  await FamilyAuthUtils.authenticateUser(
-                    context,
-                    checkAuthRequest: FamilyCheckAuthRequest(
-                      navigate: (context) async {
-                        _checkCardDetailsAndNavigate(context);
-                      },
+      );
+  Widget _inputBottomColumn() => Positioned(
+        bottom: 0,
+        child: SizedBox(
+          height: FunNumericKeyboard.getHeight(context) + 118 + 54,
+          child: Column(
+            mainAxisSize: MainAxisSize.min,
+            mainAxisAlignment: MainAxisAlignment.end,
+            children: [
+              SizedBox(
+                height: 118,
+                width: MediaQuery.of(context).size.width,
+                child: Padding(
+                  padding: const EdgeInsets.all(24),
+                  child: FunButton(
+                    onTap: widget.authcheck
+                        ? () async {
+                            await FamilyAuthUtils.authenticateUser(
+                              context,
+                              checkAuthRequest: FamilyCheckAuthRequest(
+                                navigate: (context) async {
+                                  _checkCardDetailsAndNavigate(context);
+                                },
+                              ),
+                            );
+                          }
+                        : () {
+                            _checkCardDetailsAndNavigate(context);
+                          },
+                    text: 'Give',
+                    isDisabled: !isActive,
+                    analyticsEvent: AnalyticsEvent(
+                      AmplitudeEvents.parentGiveClicked,
                     ),
-                  );
-                }
-              : () {
-                  _checkCardDetailsAndNavigate(context);
+                  ),
+                ),
+              ),
+              FunNumericKeyboard(
+                presets: const [25, 50, 75, 100],
+                onPresetTap: (amount) {
+                  setState(() {
+                    _amount = int.parse(amount);
+                    isActive = true;
+                    isPresetAmount = true;
+                  });
+                },
+                onKeyboardTap: (value) {
+                  setState(() {
+                    if (isActive) {
+                      // overwriting a previously selected preset
+                      if (isPresetAmount) {
+                        _amount = int.parse(value);
+                        isPresetAmount = false;
+                      } else {
+                        // normal typing
+                        _amount = _amount * 10 + int.parse(value);
```

Comment:
Just to be sure, what's exactly happening here?

Reply:
Nvm, i get it now!

Reply:
I think what would be helpful is to add a comment about the logic. So you're not "parsing" the amount from the edittext when the buttons are pressed but calculating in "tens".

Reply:
Added more clarifying comments 👌